lib/internal/cluster/master.js

```diff
@@ -248,19 +248,22 @@ cluster.disconnect = function(cb) {
     intercom.once('disconnect', cb);
 };
 
+const methodMessageMapping = {
+  close,
+  exitedAfterDisconnect,
+  listening,
+  queryServer,
+};
+
 function onmessage(message, handle) {
   const worker = this;
 
+  const fn = methodMessageMapping[message.act];
+
   if (message.act === 'online')
     online(worker);
-  else if (message.act === 'queryServer')
-    queryServer(worker, message);
-  else if (message.act === 'listening')
-    listening(worker, message);
-  else if (message.act === 'exitedAfterDisconnect')
-    exitedAfterDisconnect(worker, message);
-  else if (message.act === 'close')
-    close(worker, message);
+  else if (fn)
+    fn(worker, message);
 }
 
 function online(worker) {
```

Comment:
What if we just added `message` to the signature here (to appease V8's JIT) and then treated `online` as the same as the other methods to simplify things completely?

Reply:
Makes sense actually 👍  reason for not doing this was the second argument passed will be unused, so if it is not of concern i can make the changes.